Packs/TrendMicroCAS/Integrations/TrendMicroCAS/TrendMicroCAS.py

```diff
@@ -0,0 +1,523 @@
+            "mail_message_delivery_time": mail_message_delivery_time
+        }]
+
+        result = self._http_request(
+            method='POST',
+            url_suffix='mitigation/mails',
+            json_data=data
+        )
+        return result
+
+    def user_action_result_query(self, batch_id: str, start: str, end: str, limit: str) -> dict:
+        params = assign_params(batch_id=batch_id, start=start, end=end, limit=limit)
+        data = self._http_request(
+            method='GET',
+            url_suffix='mitigation/accounts',
+            params=params
+        )
+        return data
+
+    def email_action_result_query(self, batch_id: str, start: str, end: str, limit: str) -> dict:
+        params = assign_params(batch_id=batch_id, start=start, end=end, limit=limit)
+        data = self._http_request(
+            method='GET',
+            url_suffix='mitigation/mails',
+            params=params
+
+        )
+        return data
+
+    def blocked_lists_get(self):
+        result = self._http_request(
+            method='GET',
+            url_suffix='remediation/mails'
+        )
+        return result
+
+    def blocked_lists_update(self, action_type: str, senders_list: List[str], urls_list: List[str],
+                             filehashes_list: List[str]) -> dict:
+        rules = assign_params(senders=senders_list, urls=urls_list, filehashes=filehashes_list)
+        data = {
+            "action_type": action_type,
+            "rules": rules
+        }
+        result = self._http_request(
+            method='POST',
+            url_suffix='remediation/mails',
+            json_data=data
+        )
+        return result
+
+    def next_link(self, link: str) -> dict:
+        data = self._http_request(
+            method='GET',
+            full_url=link,
+            url_suffix=''
+        )
+        return data
+
+
+''' HELPER FUNCTIONS '''
+
+
+def parse_date_to_isoformat(arg: str, arg_name: str):
+    """
+        Parses date_string to iso format date strings ('%Y-%m-%dT%H:%M:%SZ'). Input Can be any date that is valid or
+        'number date range unit' for Examples: (2 hours, 4 minutes, 6 month, 1 day, etc.)
+
+        Args:
+            arg (str): The date to be parsed.
+            arg_name (str): the name of the argument for error output.
+
+        Returns:
+            str: The parsed date in isoformat strings ('%Y-%m-%dT%H:%M:%SZ').
+    """
+    if arg is None:
+        return None
+
+    # we use dateparser to handle strings either in ISO8601 format, or
+    # For example: format 2019-10-23T00:00:00 or "3 days", etc
+
+    date = dateparser.parse(arg, settings={'TIMEZONE': 'UTC'})
+    if not date:
+        return_error(f'Error in the date - {arg_name}')
+
+    date = f'{date.isoformat()}Z'
+    return date
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def test_module(client: Client) -> str:
+    try:
+        client.security_events_list(service='exchange', event_type='securityrisk')
```

Comment:
Returns an empty list of events

Reply:
do we want to always run it? or only if the IsFetch is set to `True`?
also, it seems the user can enter multiple services to fetch incidents for. I would suggest to make the test_module in that case more comprehensive.

Reply:
@bakatzir This is the "simple case" test module. @moishce is working on adding a full fetch mechanism (besides the incidents saving and the set last run) in case that the user will check the fetch_incidents checkbox.